packages/react/src/Icon/Icon.tsx

```diff
@@ -0,0 +1,34 @@
+import React, {HTMLAttributes} from 'react'
+import styles from './Icon.module.css'
+import clsx from 'clsx'
+import {Icon as IconProps} from '@primer/octicons-react'
+import {Colors} from '../constants'
+
+/**
+ * Design Tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/icon/colors.css'
+
+export const IconColors = Colors
+export const defaultIconColor = IconColors[0]
+
+type BrandIconProps = HTMLAttributes<HTMLSpanElement> & {
+  icon: IconProps
+  color?: (typeof IconColors)[number]
+  hasBackground?: boolean
+}
+
+export const Icon = ({
+  icon: BrandIcon,
+  className,
+  color = defaultIconColor,
+  hasBackground = false,
+  ...rest
+}: BrandIconProps) => (
+  <span
+    className={clsx(styles.Icon, styles[`Icon--color-${color}`], hasBackground && styles['Icon--badge'], className)}
+    {...rest}
+  >
+    <BrandIcon />
+  </span>
```

Comment:
@joshfarrant I played around with this and have ran into issues creating the background around the icon: Right now the span has the background color.

If we remove the span, we can add padding to the Octicon itself to get the background, however, that leads to inconsistent sizing between the icon with the background and the icon without (especially if we utilize the `size` prop.

I'm also happy to pass in the `size` prop but then we'll lead to splitting up props into the span and into the octicon. 

Let me know what you think.

Reply:
Thanks for looking into this @rfearing. I replied to the span question [below](https://github.com/primer/brand/pull/787).

Just wanted to clarify here — I wasn't suggesting we expose the `size` prop to consumers of the `Icon`, I was suggesting that we use that internally rather than setting the dimensions with CSS. As I've done [here](https://github.com/primer/brand/pull/787/files#diff-9716443558fe6a1655c5ba0e3fb1493d9803c5f11cd1a87e7656b71de835248bR30).

Hope that makes sense.

Reply:
<picture data-single-emoji=":think-about-it:" title=":think-about-it:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T05CSQFHRAP/think-about-it/563b2551b9775e63.png" alt=":think-about-it:" align="absmiddle"></picture> Thank makes sense. Thank you! 😄